src/yanfeishao/cs555/utils/OutputUtils.java

```diff
@@ -80,8 +81,67 @@ public void outputError(SimpleDBUtils simpleDBUtils, String prefix) {
         Set<String> result = new HashSet<>();
         switch (prefix) {
-            case ErrorCode.US01:
-                break;
+            case ErrorCode.US01: {
+                // Dates before the current date
+                Date todayDate = Calendar.getInstance().getTime();
+                simpleDBUtils.getFamilyDBList().forEach(familyEntity -> {
+                    Date marriageDate = familyEntity.getMarriedDate();
+                    Date divorceDate = familyEntity.getDivorceDate();
+                    Date husbandBirthDate = familyEntity.getFather().getBirthDate();
+                    Date wifeBirthDate = familyEntity.getMother().getBirthDate();
+                    Date husbandDeathDate = familyEntity.getFather().getDeathDate();
+                    Date wifeDeathDate = familyEntity.getMother().getDeathDate();
+                    if (husbandBirthDate != null) {
+                        if (husbandBirthDate.after(todayDate)) {
+                            result.add(String.format(FormatterRegex.ERROR_FAMILY + ErrorInfo.US01, prefix, familyEntity.getIdentifier()));
+                        }
+                    }
+                    if (husbandDeathDate != null) {
+                        if (husbandDeathDate.after(todayDate)) {
+                            result.add(String.format(FormatterRegex.ERROR_FAMILY + ErrorInfo.US01, prefix, familyEntity.getIdentifier()));
+                        }
+                    }
+                    if (wifeBirthDate != null) {
+                        if (wifeBirthDate.after(todayDate)) {
+                            result.add(String.format(FormatterRegex.ERROR_FAMILY + ErrorInfo.US01, prefix, familyEntity.getIdentifier()));
+                        }
+                    }
+                    if (wifeDeathDate != null) {
+                        if (wifeDeathDate.after(todayDate)) {
+                            result.add(String.format(FormatterRegex.ERROR_FAMILY + ErrorInfo.US01, prefix, familyEntity.getIdentifier()));
+                        }
+                    }
+                    if (marriageDate != null) {
+                        if (marriageDate.after(todayDate)) {
+                            result.add(String.format(FormatterRegex.ERROR_FAMILY + ErrorInfo.US01, prefix, familyEntity.getIdentifier()));
+                        }
+                    }
+                    if (divorceDate != null) {
+                        if (divorceDate.after(todayDate)) {
+                            result.add(String.format(FormatterRegex.ERROR_FAMILY + ErrorInfo.US01, prefix, familyEntity.getIdentifier()));
+                        }
+                    }
+                    familyEntity.getChildList().forEach(child -> {
+                        try {
+                            Date childBirthDate = child.getBirthDate();
+                            Date childDeathDate = child.getDeathDate();
+                            if (childBirthDate != null) {
```

Comment:
The same above, merge 128 - 137 into one if condition
You can use || operation, means or